security-force-monitor/sfm-cms
# Consolidate composition and membership data into a single set of spreadsheet fields

## Overview

See title.

There are two related groups of commits here:

- Updating the import and tests to accommodate the new data format: https://github.com/security-force-monitor/sfm-cms/pull/715/files/f334f64ef6671542de75c014009db3d4dfb69e5b
- Renaming and updating the data type of `MembershipOrganizationRealEnd` to `MembershipOrganizationOpenEnded` so it can accept all values for the controlled vocabulary to express open-endedness (described in https://github.com/security-force-monitor/sfm-cms/issues/141, under `Organizations` > `Parent relationships`): https://github.com/security-force-monitor/sfm-cms/pull/715/files/f334f64ef6671542de75c014009db3d4dfb69e5b..c95b2dbcf2e3df7eaad7d0696099fef03bf7d4ac

Connects #706.

### Notes

I updated the spreadsheet fields on the models but did not consolidate them in the import as suggested in the issue because we are retaining the separate models, for now, and I didn't want to break from the existing pattern, which is actually pretty nice. If we decide to make corresponding changes in the data model in the future, then we can consolidate the fields in the import at that time.

Also, there are at least three distinct updates that need to be made to the import to accommodate the new data structure, so I created a long-lived `import-refactor` branch and am drafting those changes as distinct PRs, to keep them easy to review. This is the first in that series.

## Testing Instructions

* [The unit test](https://github.com/security-force-monitor/sfm-cms/blob/8241cf8f5058e2992d6c7d35ebadec8bff0a640b/tests/test_importer.py#L160L204) should cover this case. This change represents one step toward the new format, so a full integration test (where we'd throw the new test dataset at the import) is not appropriate / possible at this stage.
